### PR TITLE
FIX : TakePOS - fix option TAKEPOS_PRODUCT_IN_STOCK must read on the POS warehouse

### PR DESCRIPTION
**Pull request by altairis-tof**
check on terminal warehouse

### CODE REVIEW

File: htdocs/takepos/ajax/ajax.php [hunk 1]
```diff
@@ -256,7 +260,9 @@
 	if (getDolGlobalInt('TAKEPOS_PRODUCT_IN_STOCK') == 1) {
 		$sql .= ' INNER JOIN '.MAIN_DB_PREFIX.'product_stock as ps';
 		$sql .= ' ON (p.rowid = ps.fk_product';
-		$sql .= " AND ps.fk_entrepot = ".((int) getDolGlobalInt("CASHDESK_ID_WAREHOUSE".$_SESSION['takeposterminal']));
+		if ( ! empty($conf->global->{'CASHDESK_ID_WAREHOUSE'.$_SESSION['takeposterminal']})) {
```
Comment (eldy) [inline htdocs/takepos/ajax/ajax.php:263]: Must use getDolGlobalString and not conf->global->{...}

Comment (altairis-tof) [reply htdocs/takepos/ajax/ajax.php]: ok it's done now

File: htdocs/takepos/ajax/ajax.php [hunk 2]
```diff
@@ -283,6 +289,10 @@
 		$sql .= $hookmanager->resPrint;
 	}
 
+	if ($conf->global->TAKEPOS_PRODUCT_IN_STOCK == 1 && empty(getDolGlobalInt('CASHDESK_ID_WAREHOUSE'.$_SESSION['takeposterminal']))) {
```
Comment (eldy) [inline htdocs/takepos/ajax/ajax.php:292]: Use
`if (getDolGlobalInt('TAKEPOS_PRODUCT_IN_STOCK') == 1`
for php8 compatibility

**Comment by altairis-tof** [comment]
@eldy it'ok now

File: htdocs/takepos/ajax/ajax.php [hunk 3]
```diff
@@ -283,6 +289,10 @@
 		$sql .= $hookmanager->resPrint;
 	}
 
+	if (getDolGlobalInt('TAKEPOS_PRODUCT_IN_STOCK') == 1 && empty(getDolGlobalInt('CASHDESK_ID_WAREHOUSE'.$_SESSION['takeposterminal']))) {
+		$sql .= ' GROUP BY p.rowid HAVING SUM(ps.reel) > 0';
```
Comment (eldy) [inline htdocs/takepos/ajax/ajax.php:293]: The select contai s rowid but also other fields.
To be compliant with sql standard and all daya ase setup, a group by must always contains all the fields in the select except the sum of course...

Comment (altairis-tof) [reply htdocs/takepos/ajax/ajax.php]: hello @eldy, i have fixed the group by.

File: htdocs/takepos/ajax/ajax.php [hunk 4]
```diff
@@ -283,6 +289,17 @@
 		$sql .= $hookmanager->resPrint;
 	}
 
+	if (getDolGlobalInt('TAKEPOS_PRODUCT_IN_STOCK') == 1 && empty(getDolGlobalInt('CASHDESK_ID_WAREHOUSE'.$_SESSION['takeposterminal']))) {
```
Comment (eldy) [inline htdocs/takepos/ajax/ajax.php:292]: empty is not allowed and useless with getDolGlobalInt

Comment (altairis-tof) [reply htdocs/takepos/ajax/ajax.php]: @eldy, ok i've fixed it

**Pull request merged**